### PR TITLE
chore(application-system): Translate country list in phone prefix dropdown

### PR DESCRIPTION
**Pull request by norda-gunni**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Enhanced phone input functionality to support multiple languages for country codes, improving localization.
	- Added support for Icelandic country codes alongside English.
	- Introduced a locale prop in the PhoneInputController to facilitate localization.

- **Bug Fixes**
	- Updated flag representation for the Netherlands Antilles in country codes.

- **Documentation**
	- Improved clarity on the handling of localization in phone input components.

### CODE REVIEW

File: libs/application/ui-fields/src/lib/PhoneFormField/PhoneFormField.tsx [hunk 1]
```diff
@@ -63,6 +64,7 @@ export const PhoneFormField: FC<React.PropsWithChildren<Props>> = ({
               ? formatText(title, application, formatMessage)
               : undefined
           }
+          locale={locale as Locale}
```
Comment (jonnigs) [inline libs/application/ui-fields/src/lib/PhoneFormField/PhoneFormField.tsx:67]: Kemur locale ekki týpað frá `useLocale()`?

File: libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts [hunk 2]
```diff
@@ -634,3 +635,307 @@ export const countryCodes = [
...
+  { name: 'Bahamaeyjar', flag: '🇧🇸', code: 'BS', dial_code: '+1242' },
+  { name: 'Bandaríkin', flag: '🇺🇸', code: 'US', dial_code: '+1' },
+  { name: 'Bandaríska Samóa', flag: '🇦🇸', code: 'AS', dial_code: '+1684' },
+  {
+    name: 'Bandarísku Jómfrúaeyjar',
+    flag: '🇻🇮',
+    code: 'VI',
+    dial_code: '+1340',
+  },
+  { name: 'Bandarísku smáeyjar', flag: '🇺🇲', code: 'UM', dial_code: '+1' },
```
Comment (jonnigs) [inline libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts:663]: "Bandarískar smáeyjar" eða "Bandarísku smáeyjarnar"

File: libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts [hunk 3]
```diff
@@ -634,3 +635,307 @@ export const countryCodes = [
...
+  { name: 'Gvæjana', flag: '🇬🇾', code: 'GY', dial_code: '+592' },
+  { name: 'Haítí', flag: '🇭🇹', code: 'HT', dial_code: '+509' },
+  {
+    name: 'Heard og McDonaldseyjar',
+    flag: '🇭🇲',
+    code: 'HM',
+    dial_code: '+672',
+  },
+  { name: 'Holland', flag: '🇳🇱', code: 'NL', dial_code: '+31' },
+  { name: 'Hollensku Antillur', flag: '🇦🇳', code: 'AN', dial_code: '+599' },
```
Comment (jonnigs) [inline libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts:739]: Vantar fána, var hann ekki réttur í EN listanum?

File: libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts (hunk 2 above)
Comment (norda-gunni) [reply libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts:663]: Takk

File: libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts (hunk 3 above)
Comment (norda-gunni) [reply libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts:739]: Mikið rétt, takk

File: libs/application/ui-fields/src/lib/PhoneFormField/PhoneFormField.tsx (hunk 1 above)
Comment (norda-gunni) [reply libs/application/ui-fields/src/lib/PhoneFormField/PhoneFormField.tsx:67]: ég þarf að taka fram að þetta sé þetta islabndis locale en ekki eitthvað annað sem TS giskar alltaf á

File: libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts [hunk 4]
```diff
@@ -634,3 +635,307 @@ export const countryCodes = [
...
+  {
+    name: 'Kongó, Lýðstjórnar-lýðveldið',
+    flag: '🇨🇩',
+    code: 'CD',
+    dial_code: '+243',
+  },
+  { name: 'Kosta Ríka', flag: '🇨🇷', code: 'CR', dial_code: '+506' },
+  { name: 'Kókoseyjar', flag: '🇨🇨', code: 'CC', dial_code: '+61' },
+  { name: 'Kólumbía', flag: '🇨🇴', code: 'CO', dial_code: '+57' },
+  { name: 'Kómorur', flag: '🇰🇲', code: 'KM', dial_code: '+269' },
```
Comment (jonnigs) [inline libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts:788]: Kannski bara ég en mér finnst Kómoreyjar hljóma aðeins betur

**Review by jonnigs — APPROVED** [review]
LGTM

File: libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts (hunk 4 above)
Comment (norda-gunni) [reply libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts:788]: Skv Árnastofnun er það kómurur 🤷

**Review by jonnigs — APPROVED** [review]

File: libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts [hunk 5]
```diff
@@ -634,3 +634,307 @@ export const countryCodes = [
...
+  { name: 'Úrúgvæ', flag: '🇺🇾', code: 'UY', dial_code: '+598' },
+  { name: 'Úsbekistan', flag: '🇺🇿', code: 'UZ', dial_code: '+998' },
+  { name: 'Vanúatú', flag: '🇻🇺', code: 'VU', dial_code: '+678' },
+  { name: 'Vatíkanið', flag: '🇻🇦', code: 'VA', dial_code: '+379' },
+  { name: 'Venesúela', flag: '🇻🇪', code: 'VE', dial_code: '+58' },
+  { name: 'Vestur-Sahara', flag: '🇪🇭', code: 'EH', dial_code: '+212' },
+  { name: 'Víetnam', flag: '🇻🇳', code: 'VN', dial_code: '+84' },
+  { name: 'Wallis- og Fútúnaeyjar', flag: '🇼🇫', code: 'WF', dial_code: '+681' },
+  { name: 'Þýskaland', flag: '🇩🇪', code: 'DE', dial_code: '+49' },
+]
```
[the remark this answers is not in the document]
Comment (norda-gunni) [reply libs/island-ui/core/src/lib/PhoneInput/countryCodes.ts]: Done

**Review by Toti91 — APPROVED** [review]

**Pull request merged**